src/tests/ftest/util/general_utils.py

```diff
@@ -288,28 +288,104 @@ def run_command(command, timeout=60, verbose=True, raise_exception=True,
         raise DaosTestError(msg)
 
 
-def run_task(hosts, command, timeout=None):
+def run_task(hosts, command, timeout=None, verbose=False):
     """Create a task to run a command on each host in parallel.
 
     Args:
-        hosts (list): list of hosts
+        hosts (object): list or NodeSet of hosts
         command (str): the command to run in parallel
         timeout (int, optional): command timeout in seconds. Defaults to None.
+        verbose (bool, optional): display message for command execution. Defaults to False.
 
     Returns:
         Task: a ClusterShell.Task.Task object for the executed command
 
     """
+    if not isinstance(hosts, NodeSet):
+        hosts = NodeSet.fromlist(hosts)
     task = task_self()
     # Enable forwarding of the ssh authentication agent connection
     task.set_info("ssh_options", "-oForwardAgent=yes")
-    kwargs = {"command": command, "nodes": NodeSet.fromlist(hosts)}
+    kwargs = {"command": command, "nodes": hosts}
     if timeout is not None:
         kwargs["timeout"] = timeout
+    if verbose:
+        log = getLogger()
+        log.info("Running on {}: {}".format(hosts, command))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions
(pylint-consider-using-f-string) Formatting a regular string which could be a f-string